globeandmail/sophi-for-wordpress
# Enhancement/341 Tracker Address config option

### Description of the Change

* In the plugin settings page's "Collector settings", added a third item called "Tracker Address".

![image](https://user-images.githubusercontent.com/25176325/198001804-29304a2b-77cb-426f-a51a-39f784b9794c.png)
[image link](https://www.screencast.com/t/c31brZr5HK)

* It defaults to "https://cdn.sophi.io/latest/sophi.min.js" as the value but can be changed by a user as desired.
* It was hardcoded in some places, so it is updated in all those places.
* URL validation is added, on an invalid entry, the default value will be set. See the screenshot:

![image](https://user-images.githubusercontent.com/25176325/198001706-5d306df4-f671-438f-b6b1-a27bd62363e3.png)
[image link](https://www.screencast.com/t/txIM2c2s1Y)


Closes #341

### Alternate Designs

screenshots added above

### Possible Drawbacks

nothing

### Verification Process

1. Visit the `Settings > Sophi.io` page in the WordPress back end.
2. In the plugin settings page's "Collector settings", see a third item called "Tracker Address".
3. Try to save a non-url value and confirm it does not save and the value saved back to the default one.
4. Confirm that the value is appended to the `assets/js/frontend/sophi-tag.js` and it is working fine. <- not sure how to test this step @jeffpaul 

### Checklist:






### Changelog Entry

> Added - A new tracker request endpoint config option.
> Removed & Changed - Hardcoded tracker endpoint removed and value used from the new tracker request endpoint setting field.

### Credits


Props @YMufleh @faisal-alvi @jeffpaul
